src/components/TodoListItem.jsx

```diff
@@ -43,4 +44,4 @@ const TodoListItem = ({ todo, onToggle, onDeleteTodo }) => {
   );
 };
 
-export default TodoListItem;
+export default React.memo(TodoListItem);
```

Comment:
창우님께서 적어주신대로 기본적으로 얕은 비교를 하고 있는데요. 
```suggestion
export default React.memo(
  TodoListItem,
  (prevProps, nextProps) => prevProps.todo === nextProps.todo
);
```
이렇게 작성하면 어떤 변화가 있을까요? 또 트레이드 오프는 어떻게 되는걸까요?

Reply:
음.. 기존에는 얕은 비교로 참조값을 비교했기 때문에, 부모 컴포넌트에서 `todoList`의 값 자체 변화가 없어도 리랜더링이 발생하면 참조 값이 변하게 되어 `TodoListItem` 컴포넌트도 리렌더링이 발생하게 되는데요. 이런 식으로 내부의 실제 값이 변경되었는지를 비교하는 깊은 비교를 하게 되면, 조금 전 처럼 단지 복사와 동일한 참조값만 바뀐 경우에도 내부 값의 변화가 없음을 인지하고 리렌더링을 막을 수 있습니다.

조금 생각해봐야할 것 같은데, 이렇게 한다면 부모에서 함수나 값등을 메모이제이션 하지 않아도 자식 컴포넌트가 리렌더링 되는 것을 제어할 수 있다는 장점이 있지만, 부모에서 메모이제이션을 사용하는 것과 사용하지 않고 리렌더링 시마다 새로 생성하는 것의 리소스 차이를 비교해봐야할 것 같습니다.

그렇지만 저의 생각은 부모에서 관리하는 함수와 값이 많을수록, 무거울수록 메모이제이션을 하는 것이 이득이고, 자식 컴포넌트에서 이렇게 깊은 비교를 하는 것 자체가 비용이 더 발생하는 것이기 때문에 지양하는 것이 좋다고 생각합니다!